src/tree/render.js

```diff
@@ -60,10 +62,10 @@ export const getSize = ({ error, warn }, element) => {
   return size;
 };
 
-export function preRenderTree(element, dataTree) {
+export function preRenderTree(element, dataTree, layout) {
```

Comment:
We tried to make preRender independent of layout, not sure we want it here. Is there a way around it?

Reply:
Yes should be possible. Thought the mode of zooming would be nice to have on the objectData. I will have a look

Reply:
Doesn't really matter as preRender is already reliant on dataTree. Just avoid adding layout as a dependency to the hook.
you could instead set it as a property on the root node, that way you'll have access to it, I did that with the warnings.

Reply:
Moved it to the root node